src/backend/resources.rs

```diff
@@ -0,0 +1,270 @@
+            CellType::Primitive {
+                name,
+                param_binding,
+                ..
+            } => match name {
+                _ if name == "std_reg" => {
+                    *count_map
+                        .entry((Id::from("std_reg"), (**param_binding).clone()))
+                        .or_insert(0) += 1
+                }
+                _ if name == "std_mem_d1" => {
+                    *count_map
+                        .entry((
+                            Id::from("std_mem_d1"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "std_mem_d2" => {
+                    *count_map
+                        .entry((
+                            Id::from("std_mem_d2"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "std_mem_d3" => {
+                    *count_map
+                        .entry((
+                            Id::from("std_mem_d3"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "std_mem_d4" => {
+                    *count_map
+                        .entry((
+                            Id::from("std_mem_d4"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d1" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d1"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d2" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d2"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d3" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d3"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d4" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d4"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ => (),
+            },
+            CellType::Component { name: _name } => todo!(),
```

Comment:
You can optionally put a string in the macro something like `todo("multi-component resource estimation is not yet supported")`. This is not vital but a good practice when slowly building out features